docker/django/Dockerfile

```diff
@@ -1,18 +1,45 @@
-FROM python:3.6
+ARG PYTHON_VERSION=3.6
+ARG IMAGE_VARIANT=slim
 
-ENV PYTHONUNBUFFERED 0
+# ==============================
+FROM helsinkitest/python:${PYTHON_VERSION}-${IMAGE_VARIANT} AS base_stage
+# ==============================
 
-RUN apt-get update \
-    && apt-get install --no-install-recommends -y gdal-bin python-gdal python3-gdal
+ENV PYTHONDONTWRITEBYTECODE true
+ENV PYTHONUNBUFFERED true
 
-COPY requirements.txt /usr/src/
+# Add tini init system https://github.com/krallin/tini
+ENV TINI_VERSION v0.18.0
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini
 
-RUN pip install --upgrade pip \
-    && pip install pip-tools \
-    && pip install --src /usr/src -r /usr/src/requirements.txt
+COPY --chown=appuser:appuser ./docker/django/docker-entrypoint.sh /app/
+ENTRYPOINT ["/tini", "--", "/app/docker/django/docker-entrypoint.sh"]
 
-RUN mkdir /code
+EXPOSE 8000
 
-WORKDIR /code
+# ==============================
+FROM base_stage AS development
+# ==============================
 
-COPY . /code
+RUN apt-install.sh gdal-bin build-essential
+
+COPY --chown=appuser:appuser requirements.txt /app/
+RUN pip --no-cache-dir install -r /app/requirements.txt
+
+COPY --chown=appuser:appuser requirements-dev.txt /app/
+RUN pip --no-cache-dir install -r /app/requirements-dev.txt
+
+COPY --chown=appuser:appuser . /app/
```

Comment:
now that you changed the owner of the files, it would be a good idea to change the user within the container:
```
USER appuser
```

Reply:
Surely. I donno why I was under the wrong assumption that `USER` was already set in the parent Dockerfile, but when I double-checked, it wasn't. Thanks for catching this, done in 
8cfd723 👍

Reply:
if it would have been set in the parent Dockerfile, then some of the `RUN` commands would not work in your Dockerfile, as they would require `root` access 🙂

Reply:
True. I even tried setting the `USER` at the top before `apt-install.sh` and it was failing with permission error :)